### PR TITLE
Add instructions for running on Pi4

### PR DESCRIPTION
**Pull request by hpgmiskin**
This PR is WIP in relation to issue #131

### CODE REVIEW

File: examples/pi4_mesa_build/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# Kompute Pi4 Mesa Build Example
+
+The Raspberry Pi 4 is an impressive little computer when you consider that the Broadcom GPU is able to run 2x 4K displays. This example intends to show how to get Vulkan Kompute up and running on a Raspberry Pi 4. This has huge potential for edge processing using the power of the Pi 4 GPU.
```
Comment (axsaucedo) [inline examples/pi4_mesa_build/README.md:3]: This is great - the way this is worded could make a really good blog post - once we merge this if you are interested in publishing one I'd be keen on adding that to the docs as well

File: examples/pi4_mesa_build/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+```
+vulkaninfo
+```
+
+## Clone Vulkan Kompute
+
+Clone Vulkan Kompute for access to the latest Python tests.
+
+```
+git clone https://github.com/EthicalML/vulkan-kompute.git
```
Comment (axsaucedo) [inline examples/pi4_mesa_build/README.md:83]: May be worth ensuring the clone is with a particular tag - eg `--branch v0.6.0` to make sure further changes don't break the setup - I'm planning to do a release soon so we can pin it to the `0.7.0` version

File: examples/pi4_mesa_build/README.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+Navigate to the available tests and install required dependencies.
+
+```
+cd vulkan-kompute/python/test
+python3 -m venv .venv
+source .venv/bin/activate
+pip install --upgrade pip wheel
+pip install -r requirements-dev.txt
+pip install git+git://github.com/EthicalML/vulkan-kompute.git@master
```
Comment (axsaucedo) [inline examples/pi4_mesa_build/README.md:96]: You can also install by running `pip install . -v` form the top  level directory

Comment (axsaucedo) [reply examples/pi4_mesa_build/README.md:96]: May also be worth installing with a tag (for now perhaps master is good enough until i do te 0.7.0 release - can do it this week)

File: examples/pi4_mesa_build/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+## Run the available tests
+
+```
+pytest
```
Comment (axsaucedo) [inline examples/pi4_mesa_build/README.md:102]: You can use --log-cli-level debug to show the vulkan logs

File: examples/pi4_mesa_build/README.md (hunk 3 above)
Comment (hpgmiskin) [reply examples/pi4_mesa_build/README.md:96]: When attempting to install from a relative path (either in the root or tests folder) I get the following error:

```
    running install
    running build
    running build_ext
    -- The C compiler identification is GNU 8.3.0
    -- The CXX compiler identification is GNU 8.3.0
    -- Check for working C compiler: /usr/bin/cc
    -- Check for working C compiler: /usr/bin/cc -- works
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Check for working CXX compiler: /usr/bin/c++
    -- Check for working CXX compiler: /usr/bin/c++ -- works
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Found Vulkan: /usr/lib/arm-linux-gnueabihf/libvulkan.so
    CMake Error at src/CMakeLists.txt:66 (add_subdirectory):
      The source directory

        /tmp/pip-req-build-flcbpbf5/external/Vulkan-Headers

      does not contain a CMakeLists.txt file.


    CMake Error at src/CMakeLists.txt:67 (get_target_property):
      get_target_property() called with non-existent target "Vulkan-Headers".


    CMake Error at src/CMakeLists.txt:78 (add_subdirectory):
      The source directory

        /tmp/pip-req-build-flcbpbf5/external/fmt

      does not contain a CMakeLists.txt file.


    CMake Error at src/CMakeLists.txt:142 (add_subdirectory):
      The source directory

        /tmp/pip-req-build-flcbpbf5/external/glslang

      does not contain a CMakeLists.txt file.


    -- Found PythonLibs: /usr/lib/arm-linux-gnueabihf/libpython3.7m.so (found version "3.7.3")
    CMake Error at python/CMakeLists.txt:2 (add_subdirectory):
      The source directory

        /tmp/pip-req-build-flcbpbf5/python/pybind11

      does not contain a CMakeLists.txt file.


    CMake Error at python/CMakeLists.txt:3 (pybind11_add_module):
      Unknown CMake command "pybind11_add_module".


    -- Configuring incomplete, errors occurred!
```

This did not occur when installing from master branch on git or `v0.6.0` in pypi.  I can investigate a little further another evening. I decided just to skirt the issue by installing from git.

File: examples/pi4_mesa_build/README.md (hunk 1 above)
Comment (hpgmiskin) [reply examples/pi4_mesa_build/README.md:3]: Yeah I would be keen to publish a blog post once everything is working and merged. There are a number of potentially interesting applications such as edge detection which I would be keen to compare performance of on the Pi CPU versus GPU.

File: examples/pi4_mesa_build/README.md (hunk 3 above)
Comment (axsaucedo) [reply examples/pi4_mesa_build/README.md:96]: No worries - I think for now best to stick with master, as there are quite a few bugs fixed, probably not worth investigating as much as it seems the latest one is the one that works

File: examples/pi4_mesa_build/README.md (hunk 2 above)
Comment (hpgmiskin) [reply examples/pi4_mesa_build/README.md:83]: Yes perhaps you are right to pin to a particular release. I will not have much time this week so I imagine `v0.7.0` will be long released by the time all tests are passing.

The other thought I was having is to test against the master branch using a Raspberry Pi. This discussion is potentially best conducted in a new issue but the basic idea would be to [host your own runner](https://docs.github.com/en/actions/hosting-your-own-runners) on a Pi and use as part of the testing suite. Very much half baked as I have not checked compatibility.

Comment (axsaucedo) [reply examples/pi4_mesa_build/README.md:83]: I think master would be probably best as I expect to do a release soon, so what compiles / works for master shoudl do for 0.7.0 as well - I think there will be some limitations like the point on no support for doubles / float64 which we'll have to ignore, as it is just a limitation on the hardware itself

File: examples/pi4_mesa_build/README.md (hunk 1 above)
Comment (axsaucedo) [reply examples/pi4_mesa_build/README.md:3]: That would be awesome

**Review by axsaucedo — APPROVED** [review]
Should be good to merge, let me know if there's anything else otherwise I can link in main docs after merge

**Comment by hpgmiskin** [comment]
Tested with the latest master `v0.7.0` with the following test output:

```
============================= test session starts =============================
platform linux -- Python 3.7.3, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
rootdir: /home/pi/vulkan-kompute
collected 12 items

test_array_multiplication.py .                                           [  8%]
test_kompute.py .....                                                    [ 50%]
test_logistic_regression.py .                                            [ 58%]
test_tensor_types.py ..s..                                               [100%]

======================== 11 passed, 1 skipped in 3.95s ========================
```

**Comment by axsaucedo** [comment]
Nice one! Looks great - thank you @hpgmiskin !

**Pull request merged**